tests/acceptance/generate-test.js

```diff
@@ -524,7 +524,15 @@ describe('Acceptance: ember generate', function() {
     return generate(['resource']).then(function() {
       expect(false).to.be.ok;
     }, function(err) {
-      expect(err.errorLog[0].message).to.equal('The `ember generate <entity-name>` command requires an entity name to be specified. For more details, use `ember help`.');
+      expect(err.name).to.equal('SilentError');
+      expect(err.message).to.equal('The `ember generate <entity-name>` command requires an entity name to be specified. For more details, use `ember help`.');
+    });
+  });
+
+  it('resource without entity name does not throw hard exception', function() {
+    return generate(['resource']).catch(function(err) {
+      expect(err.name).to.equal('SilentError');
+      expect(err.message).to.equal('The `ember generate <entity-name>` command requires an entity name to be specified. For more details, use `ember help`.');
```

Comment:
are you aware that this test will also pass if `generate(...)` resolves without an error? this is essentially the reason for the weird `expect(false).to.be.ok;` above, but if you add this, the new test will look pretty much the same as the one above.